packages/web/src/hooks/common/use-network-fee.ts

```diff
@@ -0,0 +1,146 @@
+import React from "react";
+import BigNumber from "bignumber.js";
+import { useAtomValue } from "jotai";
+
+import { GasInfo, NetworkFee, useGetEstimateGasInfo, useGetGasPrice } from "@hooks/gas";
+import { Document } from "src/types/transaction-messages.types";
+import { GasToken } from "@common/values/token-constant";
+import { useGnoswapContext } from "./use-gnoswap-context";
+import { makeEstimateGasTransaction } from "@utils/transaction-utils";
+import { WalletState } from "@states/index";
+
+export interface UseNetworkFeeReturn {
+  currentGasInfo: GasInfo | null;
+  networkFee: NetworkFee | null;
+  isLoading: boolean;
+  estimateNetworkFee: (document: Document) => Promise<{
+    currentGasInfo: GasInfo | null;
+    networkFee: NetworkFee | null;
+  }>;
+}
+
+export const useNetworkFee = (document: Document | null, gasInfo?: GasInfo | null): UseNetworkFeeReturn => {
+  const walletClient = useAtomValue(WalletState.client);
+  const { transactionService, transactionGasService } = useGnoswapContext();
+  const { data: gasPrice } = useGetGasPrice();
+
+  const {
+    data: estimatedGasInfo,
+    isLoading: isLoadingGasInfo,
+    isFetching: isFetchingGasInfo,
+  } = useGetEstimateGasInfo(document, gasInfo?.gasUsed || 0);
+
+  const currentGasInfo = React.useMemo(() => {
+    if (!estimatedGasInfo) return null;
+    return estimatedGasInfo;
+  }, [estimatedGasInfo]);
```

Comment:
I think this could work:

```suggestion
  const currentGasInfo = React.useMemo(() => estimatedGasInfo || null, [estimatedGasInfo]);
```